### PR TITLE
feat(frontend): expose new BE endpoints

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to expose BTC-related backend endpoint via the BE canister.

Note: the base branch of this PR will be changed to `main` after we merge https://github.com/dfinity/oisy-wallet/pull/2629.

### CODE REVIEW

File: src/frontend/src/lib/canisters/backend.canister.ts [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 	CustomToken,
+	PendingTransaction,
+	SelectedUtxosFeeResponse,
 	UserProfile,
 	UserToken
 } from '$declarations/backend/backend.did';
 import { idlFactory as idlCertifiedFactoryBackend } from '$declarations/backend/backend.factory.certified.did';
 import { idlFactory as idlFactoryBackend } from '$declarations/backend/backend.factory.did';
 import { getAgent } from '$lib/actors/agents.ic';
+import { backendCanisterError } from '$lib/canisters/backend.errors';
```
Comment (peterpeterparker) [inline src/frontend/src/lib/canisters/backend.canister.ts:12]: There was already a PR to introduce this `backendCanisterError`?

File: src/frontend/src/lib/canisters/backend.canister.ts [hunk 2]
```diff
@@ -94,4 +100,66 @@ export class BackendCanister extends Canister<BackendService> {
 			credential_spec: credentialSpec
 		});
 	};
+
+	btcAddPendingTransaction = async ({
+		txId,
+		network,
```
Comment (peterpeterparker) [inline src/frontend/src/lib/canisters/backend.canister.ts:106]: nitpick: you might not need to descruct everything, you might be able to use `...rest`

File: src/frontend/src/lib/canisters/backend.canister.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/canisters/backend.canister.ts:12]: It's a good idea, just not sure if we want to have a single function for all error. In ic-js we have an error parser per function.

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/canisters/backend.canister.ts:12]: It was previously done for Signer. And in a parallel PR (https://github.com/dfinity/oisy-wallet/pull/2629), I created a similar function for Backend. 

Re. the structure itself - I also have a feeling that having an error parser per function will be more usable in the future. I will adjust it for both BE and signer in the above mentioned PR. Let's review and merge that one first and then come back to this one.

Comment (peterpeterparker) [reply src/frontend/src/lib/canisters/backend.canister.ts:12]: Super. Ping me when you are ready to review again. I'll also need this new pattern to add a function to the backend api.

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/canisters/backend.canister.ts:12]: @peterpeterparker https://github.com/dfinity/oisy-wallet/pull/2629 is ready for review. I will update this one after we merge the errors parsers PR.

File: src/frontend/src/lib/canisters/backend.canister.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/canisters/backend.canister.ts]: Done.

File: src/frontend/src/lib/canisters/backend.canister.ts (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/canisters/backend.canister.ts]: @peterpeterparker aaand this PR is now also ready for re-review.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks!

It's a cool idea to map and throw the error in the canister 👍

**Pull request merged**